src/WizardsWardrobe.lua

```diff
@@ -52,6 +63,22 @@ function WW.LoadSetup( zone, pageId, index, auto, skipValidation )
 	if skipValidation == nil then skipValidation = false end
 	local setup = Setup:FromStorage( zone.tag, pageId, index )
 
+	local pageName = WW.pages[ zone.tag ][ pageId ].name
+	WW.gui.SetPanelText( zone.tag, pageName, setup:GetName() )
+	WW.currentIndex = index

```

Comment:
Moving this above the `IsEmpty` check means empty setups will appear to be loaded when they should not be. If you're dynamically creating a setup task then that might as well happen after the `IsEmpty` check too.